js/resources.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as msg from "gen/msg_generated";
+import * as flatbuffers from "./flatbuffers";
+import { assert } from "./util";
+import * as dispatch from "./dispatch";
+
+interface Resource {
+  rid: number;
+  repr: string;
+}
+
+export function resources(): Resource[] {
```

Comment:
Should we return a `{ [key: number]: string }` instead?

Reply:
Sure we can do it, then there is no need for `Resource` interface and sorting. I just took a hint from #1117 ;)

Reply:
I'm not sure actually. @kitsonk @piscisaureus @kevinkassimo any preferences?

Reply:
Discussed with @piscisaureus out of band. Let's go with `{ [key: number]: string }`